From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:34:21 -0800
Subject: [PATCH] wifi-scripts: add acs_exclude_6ghz_non_psc hostapd config option

Support excluding non-PSC channels from ACS on 6GHz radios.

Fixes: OWRT-8793

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 src/lib/netifd/hostapd.sh |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/src/lib/netifd/hostapd.sh
+++ b/src/lib/netifd/hostapd.sh
@@ -138,6 +138,7 @@ hostapd_common_add_device_config() {
 	config_add_int reg_power_type
 	config_add_boolean stationary_ap
 	config_add_int acs_num_scans
+	config_add_boolean acs_exclude_6ghz_non_psc
 
 	config_add_string acs_chan_bias
 	config_add_array hostapd_options
@@ -157,7 +158,7 @@ hostapd_prepare_device_config() {
 	json_get_vars country country3 country_ie beacon_int:100 doth require_mode legacy_rates \
 		acs_chan_bias local_pwr_constraint spectrum_mgmt_required airtime_mode cell_density \
 		rts_threshold beacon_rate rssi_reject_assoc_rssi rssi_reject_assoc_timeout rssi_ignore_probe_request \
-		maxassoc mbssid:0 band reg_power_type stationary_ap acs_num_scans
+		maxassoc mbssid:0 band reg_power_type stationary_ap acs_num_scans acs_exclude_6ghz_non_psc
 
 	hostapd_set_log_options base_cfg
 
@@ -182,6 +183,7 @@ hostapd_prepare_device_config() {
 
 	[ -n "$acs_chan_bias" ] && append base_cfg "acs_chan_bias=$acs_chan_bias" "$N"
 	[ -n "$acs_num_scans" ] && append base_cfg "acs_num_scans=$acs_num_scans" "$N"
+	[ "$acs_exclude_6ghz_non_psc" -eq "1" ] && append base_cfg "acs_exclude_6ghz_non_psc=1" "$N"
 
 	local brlist= br
 	json_get_values basic_rate_list basic_rate
